41.md

```diff
@@ -0,0 +1,71 @@
+NIP-41
+======
+
+Authentication of clients to relays
+-----------------------------------
+
+`draft` `optional` `author:Semisol`
+
+This NIP defines a way for clients to authenticate to relays by signing an ephemeral event.
+
+## Event format
+
+An event should be signed with `kind: 22241` and `content` being the WebSockets URL of the relay.
+The URL MUST:
+- Have a trailing slash if the path is `/`
+- Not have a port number if protocol is `ws` and port is `80` or protocol is `wss` and port is `443`
+- Not include the search/query
```

Comment:
This part is not clear. What would the "search/query" even be in this context? Does this line really resolve any real issue?

Reply:
Query string.

Reply:
"querystring" as in `wss://some.relay.com/?query=string` ? Is that a thing? How about `wss://some.relay.com/#tag=string` ?

Reply:
Yes, why not? I think relay URLs with querystring arguments can be used in some interesting ways. The # stuff I don't know. Is that sent to the server? I thought it was a client thing.

Reply:
Browsers interpret `#something` client side, only but of course, links can contain them, maybe to signal nip support: `wss://some.relay.nostr/#nips=1,2,3,4`

To identify towards the relay, it should be omitted for normalization.

On the other hand, `...?mirror=2` could connect you to a different machine behind the same load balancer maybe? Or, ... I don't know. Hence my question: Why do we exclude the query thing? Does this solve a known problem? I think this rule could get in the way of creative use cases and probably is confusing else.

Reply:
So you want to limit it to `protocol://domain[:port]/`, with port only being allowed for non-standard ports. Why consider the path if query isn't to be considered?

`wss://relay.minds.com/nostr/v1/ws` indeed uses a path but what's the qualitative difference to using a query string? Why allow one but not the other?

Reply:
I don't know. I think we should either do everything including query string or just the domain without scheme or port.